spec/requests/restaurants_spec.rb

```diff
@@ -0,0 +1,43 @@
+RSpec.describe 'Restaurants', type: :request do
+  context 'when logged in user tries to' do
+    login_user
+
+    describe 'GET /restaurants' do
+      context 'access restaurants list ' do
+        let!(:restaurants) { create_list(:restaurant, 5, active: true) }
```

Comment:
define this after line 4 and use only one record you don't need to create so many records it takes time and resources

Reply:
also, reuse it in other tests